### PR TITLE
Secured against PDF without any properties.

### LINKED ISSUE #23: pdDocGetInfo() crash (PDF without properties)
pdDocGetInfo() crashes when used against PDF without any properties:
```julia
(v1.0) julia> doc = pdDocOpen(filename);
(v1.0) julia> info = pdDocGetInfo(doc)
ERROR: type CosNullType has no field val
Stacktrace:
 [1] getproperty(::Any, ::Symbol) at ./sysimg.jl:18
 [2] get(::PDFIO.Cos.CosNullType) at /home/grzegorz-ubu/Dokumenty/Projekty/Julia/PDFIO.jl/src/CosObject.jl:39
 [3] pdDocGetInfo(::PDFIO.PD.PDDocImpl) at /home/grzegorz-ubu/Dokumenty/Projekty/Julia/PDFIO.jl/src/PDDoc.jl:133
 [4] top-level scope at none:0
```
Fix seems to be easy - I will send PR.

### PR DESCRIPTION
**Pull request by gwierzchowski**
Fixes #23

### CODE REVIEW

File: src/PDDoc.jl [hunk 1]
```diff
@@ -130,10 +130,12 @@ Please refer to the PDF specification for further details.
 function pdDocGetInfo(doc::PDDoc)
     obj = cosDocGetInfo(doc.cosDoc)
     dInfo = Dict{CDTextString, Union{CDTextString, CDDate}}()
-    for (key, val) in get(obj)
-        skey = CDTextString(key)
-        dInfo[skey] = (skey == "CreationDate") || (skey == "ModDate") ?
-                      CDDate(val) : CDTextString(val)
+    if obj != CosNull
```
[the remark this answers is not in the document]
Comment (sambitdash) [reply src/PDDoc.jl]: Table - 15 PDF Spec:

name| datatype | description
----|---|---
Info | dictionary |  (Optional; shall be an indirect reference) The document’s informationdictionary (see 14.3.3, "Document Information Dictionary")

**Comment by sambitdash** [comment]
Can you add a test case as well?

**Comment by gwierzchowski** [comment]
@sambitdash Hi. It looks I have little hard time with unit test, because I probably can not share PDF that produced issue for me, and all pdf writers that I checked (including program libraries) write some kind of default info record to pdf file.
I will check if i can hack python PyPDF2 to not write this info in order to produce file for test case.

**Comment by gwierzchowski** [comment]
@sambitdash I managed to do the hack in PyPDF2 and produced the file without `/Info` from one of files that you already have in test folder. But I noticed that PyPDF2 `getDocumentInfo()` returns `None` in such case (i.e. not an empty dictionary as in my patch). So I have a doubt if we maybe should also return `nothing` in such case. 
What is your opinion? 
Note that in such case we would break compatibility with older Julia versions.

**Comment by sambitdash** [comment]
> @sambitdash I managed to do the hack in PyPDF2 and produced the file without `/Info` from one of files that you already have in test folder. But I noticed that PyPDF2 `getDocumentInfo()` returns `None` in such case (i.e. not an empty dictionary as in my patch). So I have a doubt if we maybe should also return `nothing` in such case.
> What is your opinion?
> Note that in such case we would break compatibility with older Julia versions.

I personally prefer the `Nothing` return type. But pre-Julia 0.7 a function returning a `UnionAll` type or simply returning multiple types of return values was heavily penalized on performance. That's when I moved a lot of methods to actually return empty container types. This function signature was chosen that way. In Julia 1.0 the performance has been significantly improved on these cases. Hence, empty containers may not be needed to be returned. Can you test the code with `@code_warntype` in both cases and see the performance penalty. If there is none, we can move to the `Nothing` type return syntax.

File: src/PDDoc.jl [hunk 2]
```diff
@@ -129,13 +129,12 @@ Please refer to the PDF specification for further details.
 """
 function pdDocGetInfo(doc::PDDoc)
     obj = cosDocGetInfo(doc.cosDoc)
-    dInfo = Dict{CDTextString, Union{CDTextString, CDDate}}()
-    if obj != CosNull
-        for (key, val) in get(obj)
```
Comment (sambitdash) [inline src/PDDoc.jl]: Looks like your merge submission overwrote the earlier code. You may want to add it back.

**Comment by sambitdash** [comment]
You can use this file as a test file as this does not have an Info dictionary: https://github.com/sambitdash/PDFIO.jl/blob/master/test/files/spec-2.pdf

**Pull request merged**